src/sentry/services/hybrid_cloud/auth/impl.py

```diff
@@ -235,6 +237,42 @@ def get_org_ids_with_scim(
     def get_auth_providers(self, organization_id: int) -> List[RpcAuthProvider]:
         return list(AuthProvider.objects.filter(organization_id=organization_id))
 
+    def change_scim(
+        self, *, user_id: int, provider_id: int, enabled: bool, allow_unlinked: bool
+    ) -> None:
+        try:
+            auth_provider = AuthProvider.objects.get(id=provider_id)
+            user = User.objects.get(id=user_id)
+        except (AuthProvider.DoesNotExist, User.DoesNotExist):
+            return
+
+        with outbox_context(transaction.atomic(router.db_for_write(AuthProvider))):
+            auth_provider.flags.allow_unlinked = allow_unlinked
+            if auth_provider.flags.scim_enabled != enabled:
+                if enabled:
+                    auth_provider.enable_scim(user)
+                else:
+                    auth_provider.disable_scim()
+
+            auth_provider.save()
+
+    def disable_provider(self, *, provider_id: int) -> None:
+        with outbox_context(transaction.atomic(router.db_for_write(AuthProvider))):
+            try:
+                auth_provider: AuthProvider = AuthProvider.objects.get(id=provider_id)
+            except AuthProvider.DoesNotExist:
+                return
```

Comment:
should we log these failures?

Reply:
Good, general question.  They are expected from time to time -- all of our outbox handling should be "fail safe" due to consistency and timing issues in the system.  A warning is probably doable.